internal/controller/reinstall/reinstall.go

```diff
@@ -135,7 +138,11 @@ func (r *ReinstallHandler) ProcessRequest(
 	}{
 		{"Validating reinstall request", r.ensureValidReinstallRequest},
 		{"Setting reinstall request startTime", r.ensureStartTimeIsSet},
+		{"Checking preservation labels on ClusterInstance secrets", r.ensureClusterInstanceSecretsHavePreservedLabel},
+		{"Preserving data", r.ensureDataIsPreserved},
 		{"Deleting rendered manifests", r.ensureRenderedManifestsAreDeleted},
```

Comment:
I don't think we should block on this, but I wonder what would happen if we hit an error during one of these steps. For example if we did the preservation, then failed partway through deleting things, would we replace the preservation data with the new, smaller, set of things we backed up the second time through?

Reply:
An error at any stage/task will prevent the execution of the subsequent stage(s) and result in setting the `requestEndTime` and stop processing the reinstall request.